### PR TITLE
Add test for 256-color configuration values

### PR DESCRIPTION
**Pull request by yurikhan**
##### SUMMARY

See #78607.

##### ISSUE TYPE

- Test Pull Request

##### COMPONENT NAME

config

### CODE REVIEW

File: test/units/config/test_manager.py [hunk 1]
```diff
@@ -142,3 +143,16 @@ def decrypt(self, value, filename=None, obj=None):
 
         actual_value = ensure_type(vault_var, value_type)
         assert actual_value == "vault text"
+
+
+@pytest.mark.parametrize(("key", "expected_value"), (
+    ("COLOR_UNREACHABLE", "bright red"),
+    ("COLOR_VERBOSE", "rgb013"),
+    ("COLOR_DEBUG", "gray10")))
```
Comment (webknjaz) [inline test/units/config/test_manager.py:165]: If this is desired to be merged as a regression test before fixing, maybe use the xfail marker?
```suggestion
@pytest.mark.parametrize(
    ("key", "expected_value"),
    (
        ("COLOR_UNREACHABLE", "bright red"),
        pytest.param(
            "COLOR_VERBOSE", "rgb013",
            marks=pytest.mark.xfail(reason="https://github.com/ansible/ansible/issues/78607", strict=True),
        ),
        pytest.param(
            "COLOR_DEBUG", "gray10",
            marks=pytest.mark.xfail(reason="https://github.com/ansible/ansible/issues/78607", strict=True),
        ),
    ),
)
```

**Comment by bcoca** [comment]
hmm, shouldn't the test be failing?

**Comment by sivel** [comment]
This commit https://github.com/ansible/ansible/pull/78613/commits/03654a2a499b34c385efafeb3379db4c16101c38 effectively disabled the tests.

**Comment by yurikhan** [comment]
Yeah, the `xfail` mark means “expected to fail”. The fixer of the bug is supposed to remove the mark.

**Comment by yurikhan** [comment]
I assume by “my PR” you’re referring to #78617, but you accidentally the verb. Do you mean “add the commit from #78617 and revert the xfail”?

**Comment by webknjaz** [comment]
> I assume by “my PR” you’re referring to #78617, but you accidentally the verb. Do you mean “add the commit from #78617 and revert the xfail”?

I suppose so. You could first cherry-pick Brian's commit, ensure that the test starts failing (because xfail_strict produces a failure on XPASS — https://blog.ganssle.io/articles/2021/11/pytest-xfail.html). And then, do `git revert` as a finishing bit.

**Comment by webknjaz** [comment]
> This commit [03654a2](https://github.com/ansible/ansible/commit/03654a2a499b34c385efafeb3379db4c16101c38) effectively disabled the tests.

It did not. xfail is not the same as skip. It does run the test (unless you set `run=False`) and then it makes sure that it fails (`strict=True`) makes it fail when it starts passing so when the problem is actually fixed, you get notified loudly.

**Comment by bcoca** [comment]
@yurikhan yes, just add my commit/copy the code here and revert the xfail change

**Comment by mattclay** [comment]
@yurikhan Can you add a changelog fragment to document the change?

**Comment by yurikhan** [comment]
Do test suite changes deserve a changelog entry, or should I describe the bug fixed instead?

Will look into it this coming weekend.

**Comment by mattclay** [comment]
@yurikhan Only the bug fix requires a changelog entry. Unit test changes should not be in the changelog.

File: changelogs/fragments/colors.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - ansilbe terminal color settings were incorrectly limited to 16 options via 'choices', removing so all 256 can be accessed.
```
Comment (mattclay) [inline changelogs/fragments/colors.yml]:
```suggestion
  - ansible terminal color settings were incorrectly limited to 16 options via 'choices', removing so all 256 can be accessed.
```

**Review by bcoca — APPROVED** [review]

**Pull request merged**